GitObjectDb/Models/Compare/ObjectRepositoryChunkChange.cs

```diff
@@ -0,0 +1,141 @@
+using GitObjectDb.Attributes;
+using GitObjectDb.Models;
+using GitObjectDb.Reflection;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+
+namespace GitObjectDb.Models.Compare
+{
+    /// <summary>
+    /// Represents a chunk change in a <see cref="IModelObject"/> while performing a merge.
+    /// </summary>
+    [DebuggerDisplay("Property = {Property.Name}, Path = {Path}")]
+    [ExcludeFromGuardForNull]
+    public class ObjectRepositoryChunkChange
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ObjectRepositoryChunkChange"/> class.
+        /// </summary>
+        /// <param name="path">The path.</param>
+        /// <param name="property">The property.</param>
+        /// <param name="ancestor">The ancestor.</param>
+        /// <param name="ancestorValue">The ancestor value.</param>
+        /// <param name="theirs">Their node.</param>
+        /// <param name="theirsValue">Their value.</param>
+        /// <param name="ours">Our node.</param>
+        /// <param name="oursValue">Our value.</param>
+        /// <exception cref="ArgumentNullException">
+        /// path
+        /// or
+        /// property
+        /// or
+        /// ancestor
+        /// or
+        /// ancestorValue
+        /// or
+        /// theirs
+        /// or
+        /// theirsValue
+        /// or
+        /// ours
+        /// or
+        /// oursValue
+        /// </exception>
+        public ObjectRepositoryChunkChange(string path, ModifiablePropertyInfo property, JObject ancestor, JToken ancestorValue, JObject theirs, JToken theirsValue, JObject ours, JToken oursValue)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 8 parameters, which is greater than the 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS107)